src/frequenz/client/microgrid/_client.py

```diff
@@ -90,19 +90,21 @@ async def components(self) -> Iterable[Component]:
             Iterator whose elements are all the components in the microgrid.
 
         Raises:
-            ClientError: If the connection to the Microgrid API cannot be established or
-                when the api call exceeded the timeout.
+            ClientError: If the are any errors communicating with the Microgrid API,
+                most likely
+                a [GrpcStatusError][frequenz.client.microgrid.GrpcStatusError].
```

Comment:
What I'm trying to say here, is at the API level we are allowed to raise any `ClientError`, but in reality we are only raising a `GrpcStatusError` (as in the parent class), which is very important to handle server errors correctly, so to give that class more visibility. I didn't want to get into which specific errors every function might raise because any could be raised.

Maybe this would make my intentions more clear:

> If the are any errors communicating with the Microgrid API, most likely a [GrpcStatusError][frequenz.client.microgrid.GrpcStatusError] **sub-class**.

(just **emphasizing** here to make it clear what I added)